ext/fileinfo/libmagic/funcs.c

```diff
@@ -60,15 +60,14 @@ protected int
 file_printf(struct magic_set *ms, const char *fmt, ...)
 {
 	va_list ap;
-	size_t len;
 	char *buf = NULL, *newstr;
 
 	va_start(ap, fmt);
-	len = vspprintf(&buf, 0, fmt, ap);
+	vspprintf(&buf, 0, fmt, ap);
 	va_end(ap);
 
 	if (ms->o.buf != NULL) {
-		len = spprintf(&newstr, 0, "%s%s", ms->o.buf, (buf ? buf : ""));
+		spprintf(&newstr, 0, "%s%s", ms->o.buf, (buf ? buf : ""));
```

Comment:
Wouldn't this create a new warning about unused return values? If so, then perhaps we should use `php_ignore_value`. Same for the `vspprintf` call above.

Reply:
[`spprintf`](https://github.com/php/php-src/search?q=spprintf&unscoped_q=spprintf) and [`vspprintf`](https://github.com/php/php-src/search?q=vspprintf&unscoped_q=vspprintf) are used a lot but I couldn't find them used with `php_ignore_value` and there is no warning for that.

Here the warnings about unused results:
```
/home/travis/build/php/php-src/Zend/zend_execute_API.c: In function "zend_timeout_handler":
/home/travis/build/php/php-src/Zend/zend_execute_API.c:1114:4: warning: ignoring return value of "write", declared with attribute warn_unused_result [-Wunused-result]
    write(2, log_buffer, MIN(output_len, sizeof(log_buffer)));
    ^
mkdir: cannot create directory 'sapi/fpm/fpm/events/.libs': File exists
/home/travis/build/php/php-src/sapi/phpdbg/phpdbg_io.c: In function "phpdbg_output_pager":
/home/travis/build/php/php-src/sapi/phpdbg/phpdbg_io.c:211:5: warning: ignoring return value of "write", declared with attribute warn_unused_result [-Wunused-result]
     write(sock, "\r", 1);
     ^
```

Reply:
Noted. I use MSVC and we do not get such errors but I just remembered the macro from an old sqlite3 patch. In that case the PR looks good to me!